storage/buckets/buckets_test.go

```diff
@@ -272,6 +274,30 @@ func TestUniformBucketLevelAccess(t *testing.T) {
 	}
 }
 
+func TestLifecycleManagement(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	bucketName := tc.ProjectID + "-storage-buckets-tests"
+
+	if err := enableBucketLifecycleManagement(ioutil.Discard, bucketName); err != nil {
+		t.Errorf("enableBucketLifecycleManagement: %v", err)
+	}
+
+	// verify lifecycle is set
+	m, err := getBucketMetadata(ioutil.Discard, bucketName)
```

Comment:
Is there a potential delay before the rule propagates such that it will be visible? Just wondering if this could flake.

Also, seems preferable to call the library directly here rather than using another sample.

Reply:
I think metadata updates are strongly consistent: https://cloud.google.com/storage/docs/consistency

Reply:
I see the following:

```
For buckets, while metadata updates are strongly consistent for read-after-metadata-update operations, the resulting configuration changes may take time to propagate. For example, if you enable object versioning on a bucket, you should wait at least 30 seconds before deleting or overwriting objects.
```

...not sure which category this falls into? I guess the propagation from bucket configuration to individual objects is what takes a while.

Reply:
I think you're right, the metadata update itself is consistent, but the effect on objects might be delayed.

Reply:
Okay, sounds good! Will assign to you if we see a flake anytime soon though. 😸

Reply:
Can wrap with `testutil.Retry` if we're concerned.

Reply:
I think we can leave as is for now, but yes if we see a flake that's what we should do.